apis/Google.Cloud.Firestore.Data/Google.Cloud.Firestore.Data/DocumentReference.cs

```diff
@@ -59,12 +59,23 @@ internal DocumentReference(FirestoreDb database, CollectionReference parent, str
         }
 
         /// <summary>
-        /// Creates a <see cref="CollectionReference"/> for a direct child collection of this document.
+        /// Creates a <see cref="CollectionReference"/> for a child collection of this document.
         /// </summary>
-        /// <param name="collectionId">The collection ID. This must not be null or contain a forward slash.</param>
+        /// <param name="path">The path to the collection, relative to this document. Must not be null, and must contain
+        /// an odd number of slash-separated path elements.</param>
         /// <returns>A <see cref="CollectionReference"/> for the specified collection.</returns>
-        public CollectionReference Collection(string collectionId) =>
-            new CollectionReference(Database, this, PathUtilities.ValidateId(collectionId, nameof(collectionId)));
+        public CollectionReference Collection(string path)
```

Comment:
Should there be an overload that accepts a `FieldPath`? (And for Document)

Reply:
Nope, that's strictly for paths for fields within a document. A document with a subcollection isn't the same as a document with a field which is itself an object. I know it's a bit confusing :)

Reply:
OK. Glad you know how all this works ;)